api/src/services/listing.service.ts

```diff
@@ -1507,63 +1672,112 @@ export class ListingService implements OnModuleInit {
                 },
               }
             : undefined,
-          units: dto.units
-            ? {
-                create: dto.units.map((unit) => ({
-                  amiPercentage: unit.amiPercentage,
-                  annualIncomeMin: unit.annualIncomeMin,
-                  monthlyIncomeMin: unit.monthlyIncomeMin,
-                  floor: unit.floor,
-                  annualIncomeMax: unit.annualIncomeMax,
-                  maxOccupancy: unit.maxOccupancy,
-                  minOccupancy: unit.minOccupancy,
-                  monthlyRent: unit.monthlyRent,
-                  numBathrooms: unit.numBathrooms,
-                  numBedrooms: unit.numBedrooms,
-                  number: unit.number,
-                  sqFeet: unit.sqFeet,
-                  monthlyRentAsPercentOfIncome:
-                    unit.monthlyRentAsPercentOfIncome,
-                  bmrProgramChart: unit.bmrProgramChart,
-                  unitTypes: unit.unitTypes
-                    ? {
-                        connect: {
-                          id: unit.unitTypes.id,
-                        },
-                      }
-                    : undefined,
-                  amiChart: unit.amiChart
-                    ? {
-                        connect: {
-                          id: unit.amiChart.id,
-                        },
-                      }
-                    : undefined,
-                  unitAmiChartOverrides: unit.unitAmiChartOverrides
-                    ? {
-                        create: {
-                          items: unit.unitAmiChartOverrides.items,
-                        },
-                      }
-                    : undefined,
-                  unitAccessibilityPriorityTypes:
-                    unit.unitAccessibilityPriorityTypes
+          units:
+            !enableUnitGroups && dto.units
```

Comment:
question: I wonder if it would make sense to replace some of these checks for `enableUnitGroups` with throwing a `BadRequestException` at the beginning of this function if units are submitted on a jurisdiction that should have `unitGroups`